pyfolio/tears.py

```diff
@@ -1217,3 +1233,171 @@ def create_bayesian_tear_sheet(returns, benchmark_rets=None,
     plt.show()
     if return_fig:
         return fig
+
+
+@plotting_context
+def create_risk_tear_sheet(positions,
+                           style_factor_panel=None,
+                           sectors=None,
```

Comment:
Bug: what if the input data do not have the same DateTimeIndexes? It is very likely that the risk metrics data will start before and end later than the positions data. If so, the tear sheet produces misleading graphs and artificially deflated means as below. I.e. the analysis returns 0 until all data inputs have non-zero/non-nan values.

One fix is to only analyze/plot over the time range where all data inputs are non-zero/non-nan. I.e. in `create_risk_tear_sheet`, mask all data inputs to the intersection of their DateTimeIndexes. @gusgordon @twiecki thoughts?

![screen shot 2017-07-08 at 9 42 56 pm](https://user-images.githubusercontent.com/19851673/27990493-a4dd17a2-6426-11e7-87f6-7bbe4dd3a196.png)